posthog/api/prompt.py

```diff
@@ -0,0 +1,102 @@
+from typing import Any, List
+
+from dateutil import parser
+from rest_framework import authentication, exceptions, request, serializers, viewsets
+from rest_framework.decorators import action
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from posthog.api.forbid_destroy_model import ForbidDestroyModel
+from posthog.api.routing import StructuredViewSetMixin
+from posthog.auth import PersonalAPIKeyAuthentication, TemporaryTokenAuthentication
+from posthog.models.person.person import PersonDistinctId
+from posthog.models.prompt import PromptSequence, PromptSequenceState, get_active_prompt_sequences
+from posthog.permissions import ProjectMembershipNecessaryPermissions, TeamMemberAccessPermission
+
+
+class PromptSequenceSerializer(serializers.HyperlinkedModelSerializer):
+    class Meta:
+        model = PromptSequence
+        fields = ["key", "prompts", "rule", "type"]
+
+
+class PromptSequenceStateSerializer(serializers.HyperlinkedModelSerializer):
+    class Meta:
+        model = PromptSequenceState
+        fields = ["key", "last_updated_at", "step", "completed", "dismissed"]
+
+
+class PromptSequenceViewSet(StructuredViewSetMixin, ForbidDestroyModel, viewsets.ModelViewSet):
+    """
+    Create, read, update and delete prompt sequences state for a person.
+    """
+
+    serializer_class = PromptSequenceSerializer
+    permission_classes = [IsAuthenticated, ProjectMembershipNecessaryPermissions, TeamMemberAccessPermission]
+    authentication_classes = [
+        PersonalAPIKeyAuthentication,
+        TemporaryTokenAuthentication,  # Allows endpoint to be called from the Toolbar
+        authentication.SessionAuthentication,
+        authentication.BasicAuthentication,
+    ]
+
+    @action(methods=["PATCH"], detail=False)
+    def my_prompts(self, request: request.Request, **kwargs):
+        if not request.user.is_authenticated:  # for mypy
```

Comment:
Can use a test to prove it but I'd expect this to be handled by middleware not each route